### ISSUE TITLE
Update `pyproject.toml` to match Poetry 1.2

### ISSUE THREAD

**Issue by VIKTORVAV99** [electricitymaps/electricitymaps-contrib#4633]
Recently [Poetry 1.2](https://python-poetry.org/blog/announcing-poetry-1.2.0/) was released and will be the [supported version going forward](https://python-poetry.org/blog/announcing-poetry-1.2.0/#will-11-still-be-maintained-how-long-will-12-be-maintained).

While the Poetry 1.1 config works in 1.2 and vice versa (acording to the release notes) there are some changes in we should implement. Particularly relating to [dependancy groups](https://python-poetry.org/blog/announcing-poetry-1.2.0/#dependency-groups) and [optional groups](https://python-poetry.org/blog/announcing-poetry-1.2.0/#dependency-groups).

With these changes we could create one group for parsers, one for testing enviroments and maybe one for formatters (these could be left in the main group (dependencies without a group)).

This would allow us to be a little more flexible when it comes to the CI enviroment in the future as well as for local installs.

>**Note**
>Before doing any of these changes we need to make sure they are compatible with the Electricity Maps backend.

**Comment by FelixDQ** [comment]
I've tried updating to poetry 1.2.2, and it seems like our current usage of extras instead of the new dependency groups are still supported. It seems to work at least (I can still do `poetry install -E parsers`). However, it would probably be a good idea to update to the new dependency groups anyway when we have upgraded.

**Comment by VIKTORVAV99** [comment]
> I've tried updating to poetry 1.2.2, and it seems like our current usage of extras instead of the new dependency groups are still supported. It seems to work at least (I can still do `poetry install -E parsers`). However, it would probably be a good idea to update to the new dependency groups anyway when we have upgraded.

Yeah poetry 1.2.* is compatible with 1.1.* but not the other way around. So after we update the package manager we should gradually shift to the new features.

**Comment by madsnedergaard** [comment]
Has this been done? @VIKTORVAV99 :)

**Comment by VIKTORVAV99** [comment]
Nope

**Comment by marcorichetta** [comment]
I thought this issue could be closed because it was about compatibility with Poetry 1.2 and I see you're using Poetry 2.* https://github.com/electricitymaps/electricitymaps-contrib/blob/aed41ddfbcc29caeada8a1e5e54d0e5da00bb7c1/poetry.lock#L1

Then I re-read and it's about dependency groups in `pyproject.toml`, which I believe are matched.

So, **maybe**, it can be closed.

**Comment by VIKTORVAV99** [comment]
Thanks for the reminder!
